### ISSUE TITLE
java.lang.ArrayIndexOutOfBoundsException at ClientOnResponseProcessor.java:103

### ISSUE THREAD

**Issue by beston123** [seata/seata#5296]
- [x] I have searched the [issues](https://github.com/seata/seata/issues) of this repository and believe that this is not a duplicate.

### Ⅰ. Issue Description
When I perform load testing with JMeter,  got a  java.lang.ArrayIndexOutOfBoundsException in at io.seata.core.rpc.processor.client.ClientOnResponseProcessor.process(ClientOnResponseProcessor.java:103)

```
    @Override
    public void process(ChannelHandlerContext ctx, RpcMessage rpcMessage) throws Exception {
        if (rpcMessage.getBody() instanceof MergeResultMessage) {
            MergeResultMessage results = (MergeResultMessage) rpcMessage.getBody();
            MergedWarpMessage mergeMessage = (MergedWarpMessage) mergeMsgMap.remove(rpcMessage.getId());
            for (int i = 0; i < mergeMessage.msgs.size(); i++) {
                int msgId = mergeMessage.msgIds.get(i);
                MessageFuture future = futures.remove(msgId);
                if (future == null) {
                    LOGGER.error("msg: {} is not found in futures, result message: {}", msgId,results.getMsgs()[i]);
                } else {
                    future.setResultMessage(results.getMsgs()[i]);
                }
            }
        } 
```
The 'results.getMsgs()' size is 2, but 'mergeMessage.msgs.size()' is 3.
![1675417406989](https://user-images.githubusercontent.com/9409903/216572034-322d85b1-fdd9-4ff1-ac12-1552d71fe90c.jpg)

### Ⅱ. Describe what happened

```
2023-02-03 17:48:31.417 ERROR 96181 --- [ctor_RMROLE_1_1] i.s.c.rpc.netty.AbstractNettyRemoting    : 0104

java.lang.ArrayIndexOutOfBoundsException: Index 2 out of bounds for length 2
	at io.seata.core.rpc.processor.client.ClientOnResponseProcessor.process(ClientOnResponseProcessor.java:103)
	at io.seata.core.rpc.netty.AbstractNettyRemoting.processMessage(AbstractNettyRemoting.java:307)
	at io.seata.core.rpc.netty.AbstractNettyRemotingClient$ClientHandler.channelRead(AbstractNettyRemotingClient.java:419)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:324)
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:311)
	at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:432)
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:276)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:357)
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:379)
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:365)
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:719)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:655)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:581)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:493)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:989)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

Perform load testing with JMeter.
Num of threads: 25
Testing code: 
```
    @GlobalTransactional(name = "batchCreate", rollbackFor = Exception.class, timeoutMills = 60000)
    @Override
    public boolean batchCreate(boolean success) {
        Order order = new Order();
        order.setUserId(System.nanoTime());
        order.setMoney(BigDecimal.ONE);
        order.setCount(0);
        order.setProductId(0L);
        order.setStatus(0);
        orderDao.create(order);
        
        if (!success) {
            throw new RuntimeException("fail");
        }
        return true;
    }
```

### Ⅵ. Environment:

- JDK version : jdk8
- Seata version:   1.6.1
- OS : macOS 11.6
- Others:

**Comment by a364176773** [comment]
Are there any exceptions in the logs on the seata-server side at this time?

**Comment by a364176773** [comment]
https://github.com/seata/seata/issues/5279 I suspect that this problem is causing the seata-server to lose a result
You can change the value of `server.enableParallelRequestHandle` to false

**Comment by slievrly** [comment]
fix in #5282
